automation/scripts/clean_pipelines/clean_pipelines.py

```diff
@@ -0,0 +1,392 @@
+            id=experiment_id
+        ),
+        target_path=target_path,
+        item_type="experiment",
+    )
+
+
+def log_results(
+    context: mlrun.MLClientCtx,
+    deleted_items: list[tuple[str, str]],
+    failed_items: list[tuple[str, str, Exception, str]],
+    target_path: str,
+    item_type: str = "run",
+):
+    # Log results
+    context.log_result(
+        key=f"{item_type}s_deleted_count",
+        value=len(deleted_items),
+    )
+
+    # Log successfully deleted items as a dataset
+    if deleted_items:
+        df_succeeded = pd.DataFrame(deleted_items, columns=["Name", "ID"])
+        context.log_dataset(
+            key=f"{item_type}s_deleted_details",
+            df=df_succeeded,
+            target_path=target_path,
+        )
+
+    # Log the count of failed deletions
+    num_failed = len(failed_items)
+    context.log_result(key=f"{item_type}s_failed_count", value=num_failed)
+
+    # Log details of failed deletions if there are any
+    if failed_items:
+        df_failed = pd.DataFrame(
+            failed_items, columns=["Name", "ID", "Exception", "Reason"]
+        )
+        context.log_dataset(
+            key=f"{item_type}s_failed_details",
+            df=df_failed,
+            target_path=target_path,
+        )
+
+
+def perform_deletion(
+    items: list[tuple[str, str]],
+    delete_func: typing.Callable[[str], None],
+    total_items_amount: int,
+    item_type: str = "run",
+) -> tuple[list[tuple[str, str]], list[tuple[str, str, Exception, str]]]:
+    """
+    This function iterates through a list of items, attempts to delete each item using
+    the provided `delete_func`, and logs progress. It returns a list of successfully deleted
+    items and a list of items that failed to delete with associated error details.
+
+    :param items: A list of tuples, where each tuple contains the item ID and name to be deleted.
+    :param delete_func: A callable responsible for deleting each item, taking the item ID as an argument.
+    :param total_items_amount: The total number of items to delete, used for logging progress.
+    :param item_type: The type of items being deleted, used in logging messages (defaults to "run").
+
+    :return: A tuple containing:
+             - deleted_items: A list of tuples of successfully deleted items, with each tuple containing (name, ID).
+             - failed_items: A list of tuples for failed deletions, with each tuple containing
+               (name, ID, exception, exception reason).
+    """
+    deleted_count = 0
+    failed_items = []
+    deleted_items = []
+
+    for item_id, name in items:
+        try:
+            delete_func(item_id)
+            deleted_count += 1
+            deleted_items.append((name, item_id))
+            if deleted_count % 100 == 0:
+                mlrun.utils.logger.info(
+                    f"Deleted {deleted_count}/{total_items_amount} {item_type}s successfully"
+                )
+                sleep(10)
```

Comment:
why sleep?

Reply:
to avoid continuously loading kfp and sending excessive deletion requests when removing multiple runs over several hours. I do think 10 seconds is too long, we used it in Vistra cause it was the first time we ran it and wasn't sure how loading it's going to be, but I believe we can remove it or lower it, wdyt?

Reply:
5 seconds sounds good to me